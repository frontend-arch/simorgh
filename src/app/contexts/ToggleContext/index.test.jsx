import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { ComponentUsingContext } from '#testHelpers/mockComponents';

const mockToggleValues = {
  local: {
    foobarFeature: {
      enabled: true,
    },
  },
  test: {
    foobarFeature: {
      enabled: true,
    },
  },
  live: {
    foobarFeature: {
      enabled: false,
    },
  },
};

jest.mock('#lib/config/toggles', () => mockToggleValues);

// Require after mock to allow mocking of JS object
const { ToggleContext, ToggleContextProvider } = require('./index');

describe('ToggleContext', () => {
  shouldMatchSnapshot(
    `should provide a toggles object`,
    <ToggleContextProvider>
      <ComponentUsingContext context={ToggleContext} />
    </ToggleContextProvider>,
  );
});
