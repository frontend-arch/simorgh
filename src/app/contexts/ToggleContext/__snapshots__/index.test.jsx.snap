// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleContext should provide a toggles object 1`] = `
<span>
  {"toggleState":{"local":{"foobarFeature":{"enabled":true}},"test":{"foobarFeature":{"enabled":true}},"live":{"foobarFeature":{"enabled":false}}},"toggleDispatch":"*function - bound dispatchAction*"}
</span>
`;
